src/Microsoft.Azure.WebJobs.Host/Executors/FunctionInstance.cs

```diff
@@ -31,6 +33,11 @@ public Guid Id
             get { return _id; }
         }
 
+        public string TriggerDetails
```

Comment:
Having this as a dictionary, as opposed to a string, would give us a bit more flexibility when using (logging) the data as it would allow us to pass the details as distinct arguments, making this easy to query for providers supporting structured logging. It would also ensure the logging is consistent with all the different triggers.

Reply:
Yes this would be a good idea. We should be doing more structured logging wherever we can. Our App Insights logger, for example, will break those structured properties into their own queryable columns so you don't have to parse (or use Contains()) when querying.

Reply:
Could elaborate more on how we can take advantage of the structured logging in App Insights? What I'm imagining right now is that I would pass a dictionary of values through the data types, but I'd have to use a ToString() method to convert to string before logging it (a string that has the same format as the one I'm using right now). Where would we plug into the App Insights in this case?

Reply:
There isn’t anything specific to AI. Will follow up in person